pycaret/time_series/forecasting/functional.py

```diff
@@ -4,20 +4,21 @@
 import logging
 import os
 import warnings
-from typing import Any, Dict, List, Optional, Tuple, Union
+from typing import Any, Dict, List, Optional, Tuple, Union, TYPE_CHECKING
 
 import numpy as np
 import pandas as pd
 
-from sktime.forecasting.base import ForecastingHorizon
-
 from pycaret.internal.utils import check_if_global_is_not_none
 from pycaret.time_series import TSForecastingExperiment
 
+if TYPE_CHECKING:
```

Comment:
Why is this needed?

Reply:
This allows to avoid a circular dependency we would normally have. It will always be false during actual runtime, but true when handled by a 3rd party linter, allowing for type checking to work as intended.

https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING